Automattic/sensei
# Refactor Learner Courses block to use the same bar from Course Progress block

Based on https://github.com/Automattic/sensei/pull/4142
See https://github.com/Automattic/sensei/pull/4126#discussion_r602507621

### Changes proposed in this Pull Request

* It refactors the Learner Courses block to use the same bar from Course Progress block, as discussed here: https://github.com/Automattic/sensei/pull/4126#discussion_r602507621.
* In the Course Progress block, it keeps the original classes for backward compatibility.
* Since it's the same design used in the Course Progress block, the user is probably used that the counters and the bar are a single unit, so I changed the previous approach used in the "Progress bar" setting. Now it hides also the counters when it's disabled.
* Notice that it changes the original My Courses page design, but to follow the other component colors on this page, we're setting the default bar color to the current text color, as in the previous version ([video](https://github.com/Automattic/sensei/pull/4142)). cc @donnapep 
* Also, different from the original design, I had to hide the completed percentage, and decrease the counters font-size to fit well in the design (the hardest case is the 3 columns). Tested also in Spanish with 2 digits _"00 Lecciones 00 Terminados"_. It just doesn't fit in a single line on small screens with the featured image.

### Testing instructions

* Add the Course Progress block to a course.
* Customize it, and make sure it works properly in the editor and in the frontend.
* Go to a page with the Learner Courses block, customize the progress bar settings, and the colors, and make sure it works properly.


### Screenshot / Video

https://user-images.githubusercontent.com/876340/113160288-be51d500-9213-11eb-8a14-ae5ed1c5f7a9.mov
